pdq/cpp/bin/benchmark-mih.cpp

```diff
@@ -0,0 +1,205 @@
+}
+
+// ----------------------------------------------------------------
+int main(int argc, char** argv) {
+  if (argc>1 && (!strcmp(argv[1], "-h") || !strcmp(argv[1], "--help"))) {
+    usage(argv[0], 0);
+  } else {
+    do_test(argv[0], argc - 1, argv + 1);
+  }
+  return 0;
+}
+
+// ----------------------------------------------------------------
+static void usage(char* argv0, int rc) {
+  FILE* fp = (rc == 0) ? stdout : stderr;
+  fprintf(fp, "Usage: %s [options]\n", argv0);
+  fprintf(fp, "Options:\n");
+  fprintf(fp, "  -v                    Verbose output\n");
+  fprintf(fp, "  --no-timings          Disable timing output\n");
+  fprintf(fp, "  --seed N              Random seed (default: 41)\n");
+  fprintf(fp, "  --haystack-size N     Number of hashes in haystack (default: 10000)\n");
+  fprintf(fp, "  --needles-size N      Number of needle hashes (default: 1000)\n");
+  fprintf(fp, "  --distance N          Maximum Hamming distance (default: 32)\n");
+  exit(rc);
+}
+
+static void do_test(char* argv0, int argc, char** argv) {
+  int maxDistance = 32;
+  bool verbose = false;
+  unsigned int seed = 41;
+  size_t haystackSize = 10000;
+  size_t needlesSize = 1000;
+  int maxBitsToFlip = maxDistance - 1;
+
+  // Parse command line arguments
+  for (int i = 0; i < argc; i++) {
+    std::string arg = argv[i];
+    if (arg == "-v") {
+      verbose = true;
+    } else if (arg == "--seed") {
+      if (i + 1 < argc) seed = std::stoi(argv[++i]);
+    } else if (arg == "--haystack-size") {
+      if (i + 1 < argc) haystackSize = std::stoi(argv[++i]);
+    } else if (arg == "--needles-size") {
+      if (i + 1 < argc) needlesSize = std::stoi(argv[++i]);
+    } else if (arg == "--distance") {
+      maxDistance = std::stoi(argv[++i]);
+    } else {
+      usage(argv0, 1);
+      return;
+    }
+  }
+
+  // Initialize random number generator
+  std::mt19937 gen(seed);
+  
+  // Generate random hashes for needles
+  std::vector<std::pair<facebook::pdq::hashing::Hash256, std::string>> needles;
+  for (size_t i = 0; i < needlesSize; i++) {
+    auto hash = generateRandomHash(gen);
+    needles.push_back({hash, "needle_" + std::to_string(i)});
+  }
+
+  // Generate random hashes for haystack
+  std::vector<std::pair<facebook::pdq::hashing::Hash256, std::string>> haystack;
+  for (size_t i = 0; i < haystackSize; i++) {
+    auto hash = generateRandomHash(gen);
+    haystack.push_back({hash, "haystack_" + std::to_string(i)});
+  }
+
+  std::uniform_int_distribution<int> noiseDist(1, maxBitsToFlip);
+  for (const auto& needle : needles) {
+    int bitsToFlip = noiseDist(gen);
+    auto noisyHash = addNoise(needle.first, bitsToFlip, gen);
+    haystack.push_back({noisyHash, needle.second + "_noisy"});
+  }
+  std::shuffle(haystack.begin(), haystack.end(), gen);
+
+  // Build the MIH
+  std::vector<std::pair<facebook::pdq::hashing::Hash256, std::string>> matches;
+  int num_matches = 0;
+
+  std::chrono::time_point<std::chrono::system_clock> t1, t2;
```

Comment:
```suggestion
  std::chrono::time_point<std::chrono::steady_clock> t1, t2;
```

Monotonic clocks should be used for measuring time intervals. [system_clock](https://en.cppreference.com/w/cpp/chrono/system_clock) is not necessarily monotonic, but [steady_clock](https://en.cppreference.com/w/cpp/chrono/steady_clock) is.

Reply:
I'm not a moderator or official reviewer. I just saw this PR and want to help others avoid some common mistakes with std::chrono that have bitten me personally. Please defer to the moderator any suggestions.